app/src/main/java/fr/free/nrw/commons/upload/FileMetadataUtils.java

```diff
@@ -0,0 +1,44 @@
+package fr.free.nrw.commons.upload;
+
+import io.reactivex.Observable;
+import timber.log.Timber;
+
+import static androidx.exifinterface.media.ExifInterface.*;
+
+/**
+ * Support utils for EXIF metadata handling
+ *
+ */
+public class FileMetadataUtils {
+
+    /**
+     * Takes EXIF label from sharedPreferences as input and returns relevant EXIF tags
+     *
+     * @param pref EXIF sharedPreference label
+     * @return EXIF tags
+     */
+    public static Observable<String> getTagsFromPref(String pref) {
```

Comment:
Why are you returning observable instead of simple array?

Reply:
That was the original code. I saw no point in changing it, so I kept it. Probably, it was easier to iterate through tags.